File: book/petrol-pump/chapter2/section8/NozzlePrice.java
package chapter2.section8;

import pump.*;
import sodium.*;

public class NozzlePrice implements Pump
{
    public Outputs create(Inputs inputs) {
        Behavior<UpDown> nozzle1 = inputs.eNozzle1.hold(UpDown.DOWN);
        Behavior<UpDown> nozzle2 = inputs.eNozzle2.hold(UpDown.DOWN);
        Behavior<UpDown> nozzle3 = inputs.eNozzle3.hold(UpDown.DOWN);
        Lambda2<UpDown, Double, String> ifLiftedPrice =
            (u, price) ->
                u.equals(UpDown.UP) ? Formatters.formatPrice(price)
                                    : "";
        return new Outputs()
            .setPriceLCD1(
                Behavior.lift(ifLiftedPrice, nozzle1, inputs.price1))
            .setPriceLCD2(
                Behavior.lift(ifLiftedPrice, nozzle2, inputs.price2))
            .setPriceLCD3(
                Behavior.lift(ifLiftedPrice, nozzle3, inputs.price3));
    }
}

File: c#/Book/PetrolPump/PetrolPump/PetrolPump/Chapter2/Section10/NozzlePrice.cs
using System;

using PetrolPump.Pump;

using Sodium;

namespace PetrolPump.Chapter2.Section10
{

  public class NozzlePrice : IPump
  {
    public Outputs Create(Inputs inputs)
    {
      Behavior<UpDown> nozzle1 = inputs.ENozzle1.Hold(UpDown.DOWN);
      Behavior<UpDown> nozzle2 = inputs.ENozzle2.Hold(UpDown.DOWN);
      Behavior<UpDown> nozzle3 = inputs.ENozzle3.Hold(UpDown.DOWN);
      Func<UpDown, Double, String> ifLiftedPrice =
        (u, price) => u.Equals(UpDown.UP)
            ? Formatters.FormatPrice(price)
            : "";
      return new Outputs()
        .SetPriceLcd1(Behavior<UpDown>.Lift(ifLiftedPrice, nozzle1, inputs.Price1))
        .SetPriceLcd2(Behavior<UpDown>.Lift(ifLiftedPrice, nozzle2, inputs.Price2))
        .SetPriceLcd3(Behavior<UpDown>.Lift(ifLiftedPrice, nozzle3, inputs.Price3));
    }

  }
}
